scanner/nftScanner.py

```diff
@@ -32,71 +45,109 @@ def __init__(self, cf_contract_address, so_contract_address, from_block):
         self.cf_abi_file_path = '../contracts/abi/LagrangeChainlinkData.json'
         # Data NFT with single oracle contract ABI
         self.so_abi_file_path = '../contracts/abi/LagrangeChainlinkDataConsumer.json'
-        self.from_block = from_block
-        self.batch_size = 1000
 
         # DB connection
-        self.mydb = mysql.connector.connect(
-            host="localhost",
-            user=config('DB_USER'),
-            password=config('DB_PASSWORD'),
-            database='nft_data'
-        )
+        self.engine = create_engine('mysql+mysqlconnector://' + config('DB_USER') + ':' + config('DB_PASSWORD') + '@localhost/nft_data')
+        Session = sessionmaker(bind=self.engine)
+        self.session = Session()
+
         self.w3 = Web3(Web3.HTTPProvider(polygon_url))
         self.w3.middleware_onion.inject(geth_poa_middleware, layer=0)
         self.cf_abi = json.load(open(self.cf_abi_file_path))
         self.so_abi = json.load(open(self.so_abi_file_path))
-        self.mycursor = self.mydb.cursor()
+        
+        try:
+            lastScannedBlock = self.session.query(NFTData.last_scan_block).first()
+        except Exception as e:
+            logging.error("Error fetching the last_scan_block: ",e)
+        finally:
+            self.session.close()
+        # print("lastScannedBlock: ",lastScannedBlock[0])
+
+        if lastScannedBlock:
+            self.from_block = lastScannedBlock[0] + 1
+        else:
+            # Block at which the contracts were deployed
+            self.from_block = 34333512
+
+        self.batch_size = 1000
 
-        # Update owner command
-        self.update_owner_command = 'UPDATE nft_ownership SET transfer_event_block = (%s), owner_address = (%s) WHERE nft_address = (%s) AND nft_ID=(%s)'
-        # Is NFT exists check
-        self.is_nft_exists_command = 'SELECT * from nft_ownership WHERE nft_address = (%s) AND nft_ID=(%s)'
+    # Function to update the ownership of the NFT or insert a newly minted NFT
+    def update_or_insert_nft(self, blockNumber, to, contract_address, token_id):
+        try:
+            Session = sessionmaker(bind=self.engine)
+            self.session = Session()
```

Comment:
session does not need to be bonded with the class, it should be independently created and used within the function and closed with. So no need to use `self.session`.